lib/sweet_beats/melody.ex

```diff
@@ -5,17 +5,20 @@ defmodule SweetBeats.Melody do
   @microseconds_between_notes 250_000
 
   def start_link(start_time, instrument \\ Guitar, notes) do
-    GenServer.start_link(__MODULE__, {start_time, instrument, notes})
+    notes_queue = EQueue.from_list(notes)
+    GenServer.start_link(__MODULE__, {start_time, instrument, notes_queue})
```

Comment:
Thanks to @benfalk, I learned the difference between performing `GenServer` setup (such as `EQueue.from_list(notes)` here) in `start_link` vs. `init`. `start_link` happens in the originating process, while `init` happens in the `GenServer` process. So perform any memory-intensive setup in `start_link` to keep the `GenServer`’s own memory footprint small.